File: mpi_example2.py
# mpi_example1.py
from mpi4py import MPI

def main():
    comm = MPI.COMM_WORLD
    nprocs = comm.Get_size()
    my_rank = comm.Get_rank()
    print("Hello world! My rank: ", my_rank)

    # ensure that this executes after each rank has completed
    # printing Hello World
    comm.Barrier()
    if my_rank == 0:
        print("Number of processes: ", nprocs)

if __name__ == "__main__":
    main()

File: mpi_example2.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
  
  int numtasks, my_rank;

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &numtasks);
  MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
  
  printf("Hello world! My rank: %d\n", my_rank);

  // Ensure that all processes complete printing before proceeding
  MPI_Barrier(MPI_COMM_WORLD);
  
  if (my_rank == 0) {
    printf("Number of processes: %d\n", numtasks);
  }
  
  MPI_Finalize();
  return 0;
}
